app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragment.kt

```diff
@@ -138,36 +151,23 @@ class BookmarkFragment : LibraryPageFragment<BookmarkNode>(), UserInteractionHan
         super.onResume()
 
         (activity as HomeActivity).getSupportActionBarAndInflateIfNecessary().show()
-        val currentGuid = BookmarkFragmentArgs.fromBundle(requireArguments()).currentRoot.ifEmpty {
-            BookmarkRoot.Mobile.id
-        }
-
-        // Only display the sign-in prompt if we're inside of the virtual "Desktop Bookmarks" node.
-        // Don't want to pester user too much with it, and if there are lots of bookmarks present,
-        // it'll just get visually lost. Inside of the "Desktop Bookmarks" node, it'll nicely stand-out,
-        // since there are always only three other items in there. It's also the right place contextually.
-        if (currentGuid == BookmarkRoot.Root.id &&
-            requireComponents.backgroundServices.accountManager.authenticatedAccount() == null
-        ) {
-            signInView.view.visibility = View.VISIBLE
-        } else {
-            signInView.view.visibility = View.GONE
-        }
 
-        initialJob = loadInitialBookmarkFolder(currentGuid)
+        // Reload bookmarks when returning to this fragment in case they have been edited
+        val currentGuid = bookmarkStore.state.tree?.guid
+            ?: if (BookmarkFragmentArgs.fromBundle(requireArguments()).currentRoot.isNotEmpty()) {
+                BookmarkFragmentArgs.fromBundle(requireArguments()).currentRoot
```

Comment:
We should save the arguments to a local variable and remove a redundant second parsing step. The `navArgs` delegate can also make the code a bit shorter.

```suggestion
        val args by navArgs<BookmarkFragmentArgs>()
        val currentGuid = bookmarkStore.state.tree?.guid
            ?: if (args.currentRoot.isNotEmpty()) {
                args.currentRoot
```